source/docs/software/advanced-controls/system-identification/creating-routine.rst

```diff
@@ -0,0 +1,73 @@
+Creating an Identification Routine
+==================================
+
+Types of Tests
+--------------
+
+A standard motor identification routine consists of two types of tests:
+
+- **Quasistatic:** In this test, the mechanism is gradually sped-up such that the voltage corresponding to acceleration is negligible (hence, "as if static").
+- **Dynamic:** In this test, a constant 'step voltage' is given to the mechanism, so that the behavior while accelerating can be determined.
+
+Each test type is run both forwards and backwards, for four tests in total. The tests can be run in any order, but running a "backwards" test directly after a "forwards" test is generally advisable (as it will more or less reset the mechanism to its original position). ``SysIdRoutine`` provides command factories that may be used to run the tests, for example as part of an autonomous routine.
+
+User Code Setup
+---------------
+
+.. note:: Some familiarity with your language's units library is recommended and knowing how to use Consumers is required. Ths page assumes you are usng the Commands framework.
+
+To assist in creating SysId-compatible identification routines, WPILib provides the ``SysIdRoutine`` class. Users should create a ``SysIdRoutine`` object, which take both a ``Config`` object describing the test settings and a ``Mechanism`` object describing how the routine will control the relevant motors and log the measurements needed to perform the fit.
+
+Routine Config
+^^^^^^^^^^^^^^
+
+The ``Config`` object takes in a a voltage ramp rate for use in Quasistatic tests, a steady state step voltage for use in Dynamic tests, a time to use as the maximum test duration for safety reasons, and a callback method that accepts the current test state (such as "dynamic-forward") for use by a 3rd party logging solution. The constructormay be left blank to default the ramp rate to 1 volt per second and the step voltage to 7 volts.
+
+.. note:: Not all 3rd party loggers will interact with SysIdRoutine directly. Follow the directions provided with the logger you are using.
+
+The timeout and state callback are optional and defaulted to 10 seconds and null (which will log the data to a normal WPILog file) respectively.
```

Comment:
I wasn't entirely sure of what the policy around talking about vendor stuff was, plus CTRE hasnt released their SysId blog post yet as far as I know. This is the prime place to reference it though, yeah.

Reply:
It has relaxed in the past years.  As long as we are bringing useful information to the user in a fair way (sometimes easier said then done it should) it is perfectly fine.  @Daltz333 is probably the perfect person for an opinion here.

Reply:
We don't want to have step by step instructions for using external tools because those can change over time, but providing links to instructions from the vendors and saying how to use that data in our tool is fine.